Compile fail on nurb_ex1p

Hi 

Was just compiling mfem (master) on my new machine;

`gcc - v gcc version 9.3.1 20200408 (Red Hat 9.3.1-2) (GCC)`
mpich 3.3
 
``cmake -DHYPRE_DIR=$WORKDIR/hypre -DMETIS_DIR=$WORKDIR/metis -DMFEM_ENABLE_EXAMPLES=ON -DMFEM_ENABLE_MINIAPPS=ON 
 -DMFEM_THREAD_SAFE=ON -DMFEM_USE_MPI=ON 
-DMFEM_USE_OPENMP=ON -DCMAKE_INSTALL_PREFIX=..``

Build for HYPRE & METIS went fine, all of mfem with the exception of this example compiles, the failure message is 
```/home/adavis/opt/mfem/miniapps/nurbs/nurbs_ex1p.cpp: In member function ‘virtual void Diffusion2Integrator::AssembleElementMatrix(const mfem::FiniteElement&, mfem::ElementTransformation&, mfem::DenseMatrix&)’:
/home/adavis/opt/mfem/miniapps/nurbs/nurbs_ex1p.cpp:116:27: error: cannot convert ‘mfem::Vector [nd]’ to ‘mfem::Vector&’
  116 |          el.CalcShape(ip, shape);
      |                           ^~~~~
      |                           |
      |                           mfem::Vector [nd]
In file included from /home/adavis/opt/mfem/mesh/triangle.hpp:16,
                 from /home/adavis/opt/mfem/mesh/mesh_headers.hpp:21,
                 from /home/adavis/opt/mfem/mfem.hpp:42,
                 from /home/adavis/opt/mfem/bld/mfem.hpp:3,
                 from /home/adavis/opt/mfem/miniapps/nurbs/nurbs_ex1p.cpp:44:
/home/adavis/opt/mfem/mesh/../fem/fe.hpp:344:35: note:   initializing argument 2 of ‘virtual void mfem::FiniteElement::CalcShape(const mfem::IntegrationPoint&, mfem::Vector&) const’
  344 |                           Vector &shape) const = 0;
      |                           ~~~~~~~~^~~~~
/home/adavis/opt/mfem/miniapps/nurbs/nurbs_ex1p.cpp:128:40: error: ‘shape’ cannot be used as a function
  128 |                elmat(i, j) += w*shape(i)*laplace(j);
      |                                        ^
make[2]: *** [miniapps/nurbs/CMakeFiles/nurbs_ex1p.dir/build.make:83: miniapps/nurbs/CMakeFiles/nurbs_ex1p.dir/nurbs_ex1p.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3517: miniapps/nurbs/CMakeFiles/nurbs_ex1p.dir/all] Error 2
make: *** [Makefile:161: all] Error 2
```
I suppose I could comment out this example, but I was actually interested in playing with it.

--- a/miniapps/meshing/CMakeLists.txt
+++ b/miniapps/meshing/CMakeLists.txt
@@ -49,7 +49,7 @@ add_mfem_miniapp(twist
 add_test(NAME mesh-optimizer
   COMMAND mesh-optimizer -no-vis -m ${CMAKE_CURRENT_SOURCE_DIR}/icf.mesh)
 
-add_test(NAME minimal-surface COMMAND minimal-surface)
+add_test(NAME minimal-surface COMMAND minimal-surface -no-vis)
 
 # Parallel apps.
 if (MFEM_USE_MPI)
@@ -70,6 +70,6 @@ if (MFEM_USE_MPI)
 
   add_test(NAME pminimal-surface_np=4
     COMMAND ${MPIEXEC} ${MPIEXEC_NUMPROC_FLAG} ${MFEM_MPI_NP}
-    ${MPIEXEC_PREFLAGS} $<TARGET_FILE:pminimal-surface>
+    ${MPIEXEC_PREFLAGS} $<TARGET_FILE:pminimal-surface> -no-vis
     ${MPIEXEC_POSTFLAGS})
 endif()

--- a/miniapps/nurbs/nurbs_ex1.cpp
+++ b/miniapps/nurbs/nurbs_ex1.cpp
@@ -61,7 +61,7 @@ public:
       double w;
 
 #ifdef MFEM_THREAD_SAFE
-      Vector shape[nd];
+      Vector shape(nd);
       Vector laplace(nd);
 #else
       shape.SetSize(nd);

--- a/miniapps/nurbs/nurbs_ex1p.cpp
+++ b/miniapps/nurbs/nurbs_ex1p.cpp
@@ -75,7 +75,7 @@ public:
       double w;
 
 #ifdef MFEM_THREAD_SAFE
-      Vector shape[nd];
+      Vector shape(nd);
       Vector laplace(nd);
 #else
       shape.SetSize(nd);

